weather_dl/download_pipeline/parsers_test.py

```diff
@@ -565,6 +565,34 @@ def test_accepts_partition_keys_matching_sections(self):
             config = process_config(f)
             self.assertTrue(bool(config))
 
+    def test_accepts_partition_keys_not_present(self):
+        with io.StringIO(
+                """
+                [parameters]
+                dataset=foo
+                client=cds
+                target_path=bar
+                [selection]
+                month=
+                    01
+                    02
+                    03
+                year=
+                    1950
+                    1960
+                    1970
+                    1980
+                    1990
+                    2000
+                    2010
+                    2020
+                """
+        ) as f:
+            config = process_config(f)
+            params = config.get('parameters', {})
+            self.assertTrue(bool(config))
+            self.assertIsInstance(params['partition_keys'], list)
```

Comment:
@alxmrs Checking that `partition_keys` is an instance of type list even when not specified in the config file seems to be intended logic [here](https://github.com/google/weather-tools/blob/main/weather_dl/download_pipeline/parsers.py#L309) & [here](https://github.com/google/weather-tools/blob/main/weather_dl/download_pipeline/parsers.py#L339-L340)

LMK if this still holds up

Reply:
I appreciate including this in the tests. However, this is more of an implementation detail and not part of the end interface – so no need to exercise it.